pkg/cli/deployer/strategy/recreate/recreate.go

```diff
@@ -253,7 +253,7 @@ func (s *RecreateDeploymentStrategy) scaleAndWait(deployment *corev1.Replication
 	// Wait for the scale to take effect.
 	if !alreadyScaled {
 		// FIXME: This should really be a watch, however the scaler client does not implement the watch interface atm.
-		err = wait.PollImmediate(1*time.Second, retryTimeout, func() (bool, error) {
+		err = wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, retryTimeout, true, func(ctx context.Context) (bool, error) {
```

Comment:
this is unrelated to your changes, but is this `err` never handled?

Reply:
Thank you for spending extra time reviewing this PR. You are right, it is not handled and needs to be fixed. But I'll have another refactoring PR soon (in probably next H&H) and I'll fix this in there.